nexus/src/app/background/init.rs

```diff
@@ -39,128 +122,208 @@ use oximeter::types::ProducerRegistry;
+            driver: Driver::new(),
+            external_endpoints_tx,
+        };
 
-    /// task handle for deletion of database records for VMMs abandoned by their
-    /// instances.
-    pub task_abandoned_vmm_reaper: TaskHandle,
+        let background_tasks = BackgroundTasks {
+            task_internal_dns_config: Activator::new(),
+            task_internal_dns_servers: Activator::new(),
+            task_external_dns_config: Activator::new(),
+            task_external_dns_servers: Activator::new(),
+            task_metrics_producer_gc: Activator::new(),
+            task_external_endpoints: Activator::new(),
+            task_nat_cleanup: Activator::new(),
+            task_bfd_manager: Activator::new(),
+            task_inventory_collection: Activator::new(),
+            task_physical_disk_adoption: Activator::new(),
+            task_phantom_disks: Activator::new(),
+            task_blueprint_loader: Activator::new(),
+            task_blueprint_executor: Activator::new(),
+            task_crdb_node_id_collector: Activator::new(),
+            task_service_zone_nat_tracker: Activator::new(),
+            task_switch_port_settings_manager: Activator::new(),
+            task_v2p_manager: Activator::new(),
+            task_region_replacement: Activator::new(),
+            task_region_replacement_driver: Activator::new(),
+            task_instance_watcher: Activator::new(),
+            task_service_firewall_propagation: Activator::new(),
+            task_abandoned_vmm_reaper: Activator::new(),
+            task_vpc_route_manager: Activator::new(),
+
+            task_internal_dns_propagation: Activator::new(),
+            task_external_dns_propagation: Activator::new(),
+
+            external_endpoints: external_endpoints_rx,
+        };
 
-    /// task handle for propagation of VPC router rules to all OPTE ports
-    pub task_vpc_route_manager: TaskHandle,
-}
+        (initializer, background_tasks)
+    }
 
-impl BackgroundTasks {
-    /// Kick off all background tasks
+    /// Starts all the Nexus background tasks
+    ///
+    /// This function will wire up the `Activator`s in `background_tasks` to the
+    /// corresponding tasks once they've been started.
     #[allow(clippy::too_many_arguments)]
     pub fn start(
-        opctx: &OpContext,
+        self,
+        background_tasks: &'_ BackgroundTasks,
+        opctx: OpContext,
         datastore: Arc<DataStore>,
-        config: &BackgroundTaskConfig,
+        config: BackgroundTaskConfig,
         rack_id: Uuid,
         nexus_id: Uuid,
         resolver: internal_dns::resolver::Resolver,
         saga_request: Sender<SagaRequest>,
-        v2p_watcher: (
-            tokio::sync::watch::Sender<()>,
-            tokio::sync::watch::Receiver<()>,
-        ),
-        producer_registry: &ProducerRegistry,
-    ) -> BackgroundTasks {
-        let mut driver = Driver::new();
+        v2p_watcher: (watch::Sender<()>, watch::Receiver<()>),
```

Comment:
It seems like the changes in this PR should, hopefully, allow us to get rid of this (yay!). I think there's kind of a tradeoff between doing that in this PR, which gives us a nice worked example of how to use the new interface, but also makes this diff touch a bunch more files, and doing it a follow-up, which results in a much smaller diff. I don't have a particularly strong opinion here, but wanted to ask about your thoughts.

Reply:
We discussed this one offline as well, and right now, my preference is probably to get rid of it separately in the interest of merging this PR sooner and reducing merge conflict opportunities, but :woman_shrugging: it's probably not a huge deal either way.

Reply:
I do plan to look at this in a follow-up.